### PR TITLE
Data catalog and shown data

### PR DESCRIPTION
**Pull request by b4l**
The tick in the catalog is missing and the geocat link / legend throws an error. Also loading the cross sections that change to the hydrological bacckground map results in strange opacity settings of the the base map.

### CODE REVIEW

File: src/elements/ngm-map-configuration.js [hunk 1]
```diff
@@ -24,46 +27,62 @@ class NgmMapConfiguration extends LitElementI18n {
...
   render() {
     return html`
-      <ngm-map-chooser @change="${(event) => this.updateTransparency(event.detail.active)}"></ngm-map-chooser>
-      <div class="ngm-displayed-container">
-        <label>${i18next.t('dtd_transparency_label')}</label>
-        <div class="ui grey small slider"></div>
+      <ngm-map-chooser></ngm-map-chooser>
+      <div class="ui divider"></div>
+      <div class="ngm-layer-icon ${classMap({'ngm-visible-icon': this.visible, 'ngm-invisible-icon': !this.visible})}"
+            @click=${this.changeVisibility}></div>
```
[the remark this answers is not in the document]
Comment (b4l) [reply src/elements/ngm-map-configuration.js]: done!

File: src/style/icons.css [hunk 2]
```diff
@@ -120,9 +120,51 @@
   -webkit-mask: url('./images/i_cam_tilt.svg') no-repeat center;
 }
 
-.ngm-zoom-plus-icon {
-  width: 20px;
-  height: 20px;
```
Comment (b4l) [inline src/style/icons.css:125]: @vladyslav-tk sorry messed this up solving the merge conflicts. The issue is that we both specified the zoom icon with different name and size. I think we should only have it once. I prefer your naming and my size what do you think?

File: src/layers/ngm-layers.ts [hunk 3]
```diff
@@ -1,46 +1,38 @@
 import {html} from 'lit';
+import {customElement, property} from 'lit/decorators.js';
 import {repeat} from 'lit/directives/repeat.js';
-
 import {LitElementI18n} from '../i18n.js';
+import {Config} from './ngm-layers-item';
+// eslint-disable-next-line no-duplicate-imports
+import './ngm-layers-item';
```
Comment (vladyslav-tk) [inline src/layers/ngm-layers.ts:7]: maybe better to move `Config` interface to `layertree.ts`

File: src/style/layers.css [hunk 4]
```diff
@@ -1,92 +1,207 @@
...
+
+/* Hide the browser's default checkbox */
+.ngm-catalog-entry input {
+  position: absolute;
+  opacity: 0;
+  height: 0;
+  width: 0;
+}
+
+/* Custom checkbox */
```
Comment (vladyslav-tk) [inline src/style/layers.css:165]: I also added a custom checkbox (https://github.com/swissgeol/ngm/blob/new_ui/src/style/ngm-checkbox.css), we should merge it somehow

File: src/style/ngm-side-bar.css [hunk 5]
```diff
@@ -70,12 +70,19 @@ ngm-auth:hover {
   position: absolute;
   margin-left: 80px;
   z-index: 1;
+  display: flex;
+  flex-direction: column;
 }
```
Comment (vladyslav-tk) [inline src/style/ngm-side-bar.css:76]: To have shadow on extension panel:
```suggestion
.ngm-side-bar-panel.ngm-layer-catalog {
z-index: 2;
}
```
Also, need to increase the z-index of the header to keep the shadow on this panel.

File: src/elements/ngm-side-bar.js [hunk 6]
```diff
@@ -301,7 +296,7 @@ class SideBar extends LitElementI18n {
 
       // todo remove after panel UI update
```
Comment (vladyslav-tk) [inline src/elements/ngm-side-bar.js:297]: we can remove todo

**Review by vladyslav-tk — APPROVED** [review]
Looks good, please fix small comments

File: src/style/icons.css (hunk 2 above)
Comment (vladyslav-tk) [reply src/style/icons.css:125]: we can change size depending on the component. Currently, seems like we have this icon in object info and in the data catalog, according to the mockup, in both places size should be 20x20px

Comment (b4l) [reply src/style/icons.css:125]: ok, converted to your styling/naming

File: src/layers/ngm-layers.ts (hunk 3 above)
Comment (b4l) [reply src/layers/ngm-layers.ts:7]: Config is an extension of the `LayerTreeNode` Interface in `layertree.js` with some lit element respective ux specific config stuff. We could merge them but i would prefer not to to separate somewhat the data config from the ui/ux config.

File: src/layers/helpers.ts [hunk 7]
```diff
@@ -63,16 +65,16 @@ export function create3DTilesetFromConfig(viewer, config, tileLoadCallback) {
     }
     tileset.style = new Cesium3DTileStyle(config.style);
   }
-  tileset.pickable = config.pickable !== undefined ? config.pickable : false;
+  // FIXME: wtf
+  // tileset.pickable = config.pickable !== undefined ? config.pickable : false;
```
[the remark this answers is not in the document]
Comment (b4l) [reply src/layers/helpers.ts:69]: ok, reverted it

File: src/style/layers.css (hunk 4 above)
Comment (b4l) [reply src/style/layers.css]: done in https://github.com/swissgeol/ngm/pull/653/commits/e284f40cf4a345fcc1eea44f833a3ba24d648f6f

**Comment by b4l** [comment]
@vladyslav-tk tried to address your feedback, there is still one open remark. it was quite tedious to keep up with the `new_ui` branch. also it has some conflicts with the tooling rework. can you have a look and merge if it is fine with you?

File: src/elements/ngm-side-bar.js [hunk 8]
```diff
@@ -128,14 +128,14 @@ class SideBar extends LitElementI18n {
...
-                      @layerclick=${evt => this.onCatalogLayerClicked(evt)}>
+                     @layerclick=${evt => this.onCatalogLayerClicked(evt)}>
         </ngm-catalog>
       </div>
       <div .hidden=${this.activePanel !== 'tools'} class="ngm-side-bar-panel">
         <div .hidden=${!this.showHeader} class="ngm-panel-header">${i18next.t('lsb_tools')}
           <div class="ngm-close-icon" @click=${() => this.activePanel = ''}></div>
         </div>
-        <ngm-tools></ngm-tools>
+        <ngm-tools .hidden=${this.activePanel !== 'tools'} @close=${() => this.activePanel = ''}></ngm-tools>
```
Comment (b4l) [inline src/elements/ngm-side-bar.js:138]: @vladyslav-tk this is already done in the parent div

Comment (vladyslav-tk) [reply src/elements/ngm-side-bar.js]: updated

**Pull request merged**